File: 1186-maximum-subarray-sum-with-one-deletion/1186-maximum-subarray-sum-with-one-deletion.cpp
class Solution {
public:
    int maximumSum(vector<int>& arr) {
        int n = arr.size();
        int currMaxSum = arr[0];
        int maxSum  = arr[0];
        int oneDelete = 0;

        for(int i =1; i<n; i++){
            oneDelete = max(oneDelete+arr[i], currMaxSum);
            currMaxSum = max(currMaxSum+arr[i], arr[i]);
            maxSum = max(maxSum, max(oneDelete, currMaxSum));
        }
        
        return maxSum;
    }
};

File: 1186-maximum-subarray-sum-with-one-deletion/1186-maximum-subarray-sum-with-one-deletion.py
class Solution:
    def maximumSum(self, arr: List[int]) -> int:
        n = len(arr)
        currMaxSum = arr[0]
        maxSum = arr[0]
        oneDelete = 0

        for i in range(1, n):
            oneDelete = max(oneDelete+arr[i], currMaxSum)
            currMaxSum = max(currMaxSum+arr[i], arr[i])
            maxSum = max(maxSum, max(oneDelete, currMaxSum))

        return maxSum
        